File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Projection/EnrollmentList/Enrollment.cs
using MongoDB.Bson.Serialization.Attributes;

namespace CreditCardEnrollment.CreditCard.Enrollment.Projection.EnrollmentList;

public class Enrollment {
    [BsonId]
    public required string Id { get; init; }
    public required string UserId { get; init; }
    public required string ProductId { get; init; }
    public required DateTime RequestedDate { get; init; }
    public required string Status { get; set; }
    public string? StatusReason { get; set; }
    public DateTime? ReviewedOn { get; set; }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/enrollment/projection/enrollmentList/Enrollment.ts
export class Enrollment {
    constructor(
        // needs to be _id to be recognized as an _id field by MongoDB
        public readonly _id: string,
        public readonly userId: string,
        public readonly productId: string,
        public readonly requestedDate: Date,
        public status: string,
        public statusReason?: string,
        public reviewedOn?: Date
    ) {}
}